### PR TITLE
ref(react-router): Normalize react-router/lib/* imports

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Don't use `/lib` imports.

Replaces many `Params` with `RouteComponentProps`

### CODE REVIEW

File: static/app/views/performance/traceDetails/index.tsx [hunk 1]
```diff
@@ -21,11 +20,9 @@ import withOrganization from 'app/utils/withOrganization';
 
 import TraceDetailsContent from './content';
 
-type Props = {
+type Props = RouteComponentProps<{traceSlug: string}, {}> & {
```
Comment (priscilawebdev) [inline static/app/views/performance/traceDetails/index.tsx:23]:
```suggestion
type Props = Pick<RouteComponentProps<{traceSlug: string}, {}>, 'params' | 'location'> & {
```

File: static/app/views/performance/transactionDetails/index.tsx [hunk 2]
```diff
@@ -14,10 +13,8 @@ import withOrganization from 'app/utils/withOrganization';
 import EventDetailsContent from './content';
 import FinishSetupAlert from './finishSetupAlert';
 
-type Props = {
+type Props = RouteComponentProps<{eventSlug: string}, {}> & {
```
Comment (priscilawebdev) [inline static/app/views/performance/transactionDetails/index.tsx:16]:
```suggestion
type Props = Pick<RouteComponentProps<{eventSlug: string}, {}>, 'params' | 'location'> & {
```

File: static/app/views/performance/vitalDetail/index.tsx [hunk 3]
```diff
@@ -26,15 +25,13 @@ import {addRoutePerformanceContext, getTransactionName} from '../utils';
 
 import VitalDetailContent from './vitalDetailContent';
 
-type Props = {
+type Props = RouteComponentProps<{}, {}> & {
   api: Client;
   location: Location;
```
Comment (priscilawebdev) [inline static/app/views/performance/vitalDetail/index.tsx:30]: should we remove location and maybe use Pick here too? :D

```suggestion
```

File: static/app/views/performance/vitalDetail/index.tsx [hunk 4]
```diff
@@ -26,15 +25,13 @@ import {addRoutePerformanceContext, getTransactionName} from '../utils';
 
 import VitalDetailContent from './vitalDetailContent';
 
-type Props = {
+type Props = RouteComponentProps<{}, {}> & {
```
Comment (priscilawebdev) [inline static/app/views/performance/vitalDetail/index.tsx:28]:
```suggestion
type Props = Pick<RouteComponentProps<{}, {}>, 'location' | 'router'> & {
  api: Client;
  organization: Organization;
  projects: Project[];
  selection: GlobalSelection;
  loadingProjects: boolean;
};
```

File: static/app/views/performance/trends/index.tsx [hunk 5]
```diff
@@ -19,13 +18,11 @@ import {generatePerformanceEventView} from '../data';
 
 import TrendsContent from './content';
 
-type Props = {
+type Props = RouteComponentProps<{}, {}> & {
```
Comment (priscilawebdev) [inline static/app/views/performance/trends/index.tsx:21]:
```suggestion
type Props = Pick<RouteComponentProps<{}, {}>, 'location'> & {
```

File: static/app/views/performance/transactionSummary/index.tsx [hunk 6]
```diff
@@ -45,10 +44,8 @@ import {
 import {ZOOM_END, ZOOM_START} from './latencyChart';
 import {TransactionThresholdMetric} from './transactionThresholdModal';
 
-type Props = {
+type Props = RouteComponentProps<{}, {}> & {
```
Comment (priscilawebdev) [inline static/app/views/performance/transactionSummary/index.tsx:47]:
```suggestion
type Props = Pick<RouteComponentProps<{}, {}>, 'location'> & {
```

File: static/app/views/organizationJoinRequest.tsx [hunk 7]
```diff
@@ -11,9 +11,7 @@ import {trackAdhocEvent} from 'app/utils/analytics';
 import EmailField from 'app/views/settings/components/forms/emailField';
 import Form from 'app/views/settings/components/forms/form';
 
-type Props = {
-  params: Params;
-};
+type Props = RouteComponentProps<{orgId: string}, {}>;
```
Comment (priscilawebdev) [inline static/app/views/organizationJoinRequest.tsx:14]:
```suggestion
type Props = Pick<RouteComponentProps<{orgId: string}, {}>, 'params'>;
```

File: static/app/views/eventsV2/landing.tsx [hunk 8]
```diff
@@ -42,10 +41,8 @@ const SORT_OPTIONS: SelectValue<string>[] = [
 
 type Props = {
   organization: Organization;
-  location: Location;
-  router: ReactRouter.InjectedRouter;
-  params: Params;
-} & AsyncComponent['props'];
+} & RouteComponentProps<{}, {}> &
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/landing.tsx:44]:
```suggestion
} & Pick<RouteComponentProps<{}, {}>, 'location'> &
```

Comment (EvanPurkhiser) [reply static/app/views/eventsV2/landing.tsx:44]: It needs `router` and `params` as well here.

I realize I am kindof inconsistent about pick, but I'm more concerned about just using the same `RouteComponnetProps` everywhere.

File: static/app/views/performance/vitalDetail/index.tsx (hunk 3 above)
Comment (EvanPurkhiser) [reply static/app/views/performance/vitalDetail/index.tsx]: Removed `location` 👍

**Comment by EvanPurkhiser** [comment]
@priscilawebdev I think I am going to save the `Pick` cleanup for another refactoring audit. For this one I just wanted to stop using things like `Params`.

if you actually look at our imports in react-router after this, there's a number of types that are no longer imported.

I have another follow up that makes usage of `WithRouterProps` more consistent as well.

Was there anything other than the `Pick` suggestion I missed? If not I think this should hopefully be good to go

**Review by priscilawebdev — APPROVED** [review]
lgtm 👏👏

**Pull request merged**